### DISCUSSION TITLE
What's the minimum RAM required for ShredOS to boot

### DISCUSSION THREAD

**Discussion by Firminator** [PartialVolume/shredos.x86_64#99] [Q&A]
I have an old box (440BX Pentium III) and it came with 1 stick of 64MB PC133 RAM.
ShredOS tries to boot, but gives an error message right away complaining about RAM.

**Comment by Firminator** [comment]
"error: out of memory."

**Comment by PartialVolume** [comment]
For the 32 bit version of ShredOS you will need about 112MB without running any wipes. So I would say you are looking at a minimum of 128MB memory to boot ShredOS and also have sufficient memory to create multiple wipes.

The 32bit of ShredOS uses the i686 architecture variant so should run on everything from Pentium 2 and newer.

**Regarding x86_64 (64bit) builds of ShredOS.** Other specific architectures that ShredOS can be built too include core2, corei7, westmere, corei7-avx, core-avx2, atom, silvermont, opteron, opteron w/ SSE3, barcelona, jaguar, steamroller. nocona is the lowest common denominator that will run on most processors.

**Comment by PartialVolume** [comment]
One further comment to correct the terminology I used. nocona, core2, corei7 should have been refereed to as architecture variants. i.e variants of the x86_64 and not applicable to the Pentium3

Varients of the i386 (32bit) architecture include nocono, i486, i586, x1000, i686, pentium pro, pentium MMX, pentium mobile, pentium 2, pentium3, pentium4, prescott, nocona, core2, corei7, westmere, corei7-avx, core-avx2, atom, silvermont, k6, k6-2, athlon, athlon-4, opteron, opteron w/SSE3, barcelona, jaguar, steamroller, geode, Via/Cyrix C3 (Samual/Ezra cores)

**[Edit] However I believe I use the i686 variant when compiling for i386 (32bit) build so the current build of 32bit ShredOS may work fine on a Pentium3 as long as there is a minimum of 128MB of CPU memory.**

**Comment by Firminator** [comment]
Thanks. I'll try to get two 128MB sticks and then continue testing to see if the PIII is supported. If not, I'll might be able to try out your recent ShredOS build instructions to build me a custom image with PIII support. You think this could be done for a first timer trying to compile ShredOS?
Unfortunately I have the box at work so I can't test the builds over the Easter break.

**Reply by PartialVolume** [reply]
Yes, those build instructions should be fine. To change the architecture & variant is dead easy. Run `make menuconfig`, select `target options` and then `target architecture` and `target variant`.
![Screenshot_20220412_195240](https://user-images.githubusercontent.com/22084881/163033336-cbc49672-44cb-42b6-b183-30993f820593.png)
![Screenshot_20220412_195428](https://user-images.githubusercontent.com/22084881/163033411-437829de-a632-4402-b8fb-2eec7323dff9.png)
![Screenshot_20220412_195547](https://user-images.githubusercontent.com/22084881/163033616-a8c8fab7-6c09-41a3-83bc-acd52ac62569.png)
![Screenshot_20220412_195623](https://user-images.githubusercontent.com/22084881/163033738-01ae0cad-dda9-4665-aa3a-fbab5d216a0e.png)

**Reply by PartialVolume** [reply]
I'm pretty sure I build using the i686 variant for the i386 (32bit) architecture. I use the nocona variant for x86_64 architecture. So I think the current build of ShredOS 32bit should run on a Pentium 3 as long as you have a minimum of 128MB of CPU memory.

Will be interesting to see if it works! :-)

**Comment by PartialVolume** [comment]
@Firminator I've updated my comments regarding which architecture variants I'm using for 32 and 64 bit builds, as having re-read them, they needed clarification and correction. If you are confused by the github emails which contain incorrect information regarding which variant I'm using for 32 bit builds then refer to the github discussion page themselves which I have edited to hopefully make things a little clearer. 

So in summary:
64 bit build uses the nocona variant and will run on everything from Pentium 4 and newer.
32 bit build uses the i686 variant and will run on everything from Pentium 2 and newer.

ShredOS can be built for earlier processors right back to 486 but they must have a minimum of 128MB of CPU memory.
